src/orion/algo/robo/bohamiann.py

```diff
@@ -79,48 +94,50 @@ class RoBO_BOHAMIANN(RoBO):
 
     def __init__(
         self,
-        space,
-        seed=0,
-        n_initial_points=20,
-        maximizer="random",
-        acquisition_func="log_ei",
-        normalize_input=True,
-        normalize_output=False,
-        burnin_steps=None,
-        sampling_method="adaptive_sghmc",
-        use_double_precision=True,
-        num_steps=None,
-        keep_every=100,
-        learning_rate=1e-2,
-        batch_size=20,
-        epsilon=1e-10,
-        mdecay=0.05,
-        verbose=False,
+        space: Space,
+        seed: int | Sequence[int] | None = 0,
+        n_initial_points: int = 20,
+        maximizer: MaximizerName = "random",
+        acquisition_func: AcquisitionFnName = "log_ei",
+        normalize_input: bool = True,
+        normalize_output: bool = False,
+        burnin_steps: bool | None = None,
+        sampling_method: SamplingMethod = "adaptive_sghmc",
+        use_double_precision: bool = True,
+        num_steps: int | None = None,
+        keep_every: int = 100,
+        learning_rate: float = 1e-2,
+        batch_size: int = 20,
+        epsilon: float = 1e-10,
+        mdecay: float = 0.05,
+        verbose: bool = False,
     ):
 
-        super(RoBO_BOHAMIANN, self).__init__(
+        super().__init__(
             space,
             maximizer=maximizer,
             acquisition_func=acquisition_func,
-            normalize_input=normalize_input,
-            normalize_output=normalize_output,
-            burnin_steps=burnin_steps,
-            sampling_method=sampling_method,
-            use_double_precision=use_double_precision,
-            num_steps=num_steps,
-            keep_every=keep_every,
-            learning_rate=learning_rate,
-            batch_size=batch_size,
-            epsilon=epsilon,
-            mdecay=mdecay,
-            verbose=verbose,
             n_initial_points=n_initial_points,
             seed=seed,
         )
+        self.normalize_input = normalize_input
```

Comment:
We do they need to be set outside of `__init__` now? Won't they be missed when creating the configuration dict? (not be part of `param_names`)

Reply:
No since they are in the signature of the function, they will get picked up by the `inspect.signature(type(self).__init__)`, which works on the current class, not just the base class. 
The configuration dict keys will be all the arguments to `RoBO_BOHAMIANN.__init__`, if they are also an attribute on `self`.

Reply:
Ok so here's the deal. The `seed` needs to be used to create the models, and also to set something on the models. So `seed_rng` would have to do something both before, and after the `model` and `robo` attributes are set.

Reply:
Nevermind, forget most of what I said before, I just realized why this is here.

The problem is, `build_model` needs to be called eventually, and it may depend on some attributes that are set on `self` in subclasses. Therefore, it can't be called by the base class, but it has to be called at some point!

Reply:
Oh right, I forgot about these changes. Thanks!